src/xrpld/app/tx/detail/LoanPay.cpp

```diff
@@ -305,7 +305,13 @@ LoanPay::doApply()
     // change will be discarded.
     if (loanSle->isFlag(lsfLoanImpaired))
     {
-        LoanManage::unimpairLoan(view, loanSle, vaultSle, asset, j_);
+        if (auto const ret =
+                LoanManage::unimpairLoan(view, loanSle, vaultSle, asset, j_);
+            ret != tesSUCCESS)
```

Comment:
You don't need to explicitly check `ret`.

```suggestion
        if (auto const ret =
LoanManage::unimpairLoan(view, loanSle, vaultSle, asset, j_))
```